### ISSUE TITLE
Support for truncation_strategy, max_prompt_tokens and max_completion_tokens

### ISSUE THREAD

**Issue by passivebook** [openai/openai-python#1312]
### Confirm this is a feature request for the Python library and not the underlying OpenAI API.

- [X] This is a feature request for the Python library

### Describe the feature or improvement you're requesting

Can the python package be updated to support truncation_strategy, max_prompt_tokens and max_completion_tokens as seen in the [OpenAI Documentation here](https://platform.openai.com/docs/api-reference/runs/createRun#runs-createrun-truncation_strategy)?

Is it possible to do this already?

**Comment by CoderTonyB** [comment]
I'm having the same problem.  I need to upload a few thousand words as part of my workflow and it's truncating them.  I see truncation_strategy as a header, but I can't figure out where to add them.  As shown here:
https://platform.openai.com/docs/api-reference/runs/createRun#runs-createrun-truncation_strategy

I've tried using extra_headers:

```python
assistant = client.beta.assistants.create(
    name="PDF Extract-O-Matic",
    instructions=instruction,
    tools=[{"type": "code_interpreter"}],
    model="gpt-4-turbo-2024-04-09",
    extra_headers={"truncation_strategy": "last_messages", "last_messages": "10"},
)
 but when I add a message it still truncates the middle.  
 
 My workaround right now is just to upload the larger text as a file and reference it.  It would be ideal if I didn't have to do that.

**Comment by rattrayalex** [comment]
I believe `truncation_strategy` is a body parameter, not a header. 

This will be added to the SDKs shortly; thank you for your patience.

**Comment by pstern-sl** [comment]
This is available in version 1.18 released today https://github.com/openai/openai-python/pull/1315
